chapter05/ard/ex2.pl

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use 5.010;
+
+my %data = ();
+
+# Bit of a hack - I'm using this as a fake "destination" host to count total traffic
+# from each source. Contains asterisks and spaces so it shouldn't ever clash with a real host name.
+my $any_dest = '* ANYWHERE *';
```

Comment:
Agreed.  I was tempted to make an uber-hash with sub-arrayrefs and hashrefs to contain everything but thought better of it.
Amusingly, the solutions did something similar to this.